### PR TITLE
Fixing JacocoRunner building script for Jacoco

### PR DESCRIPTION
**Pull request by gergelyfabian**
- Latest Bazel uses Jacoco 0.8.7
- Ensure that for building JacocoRunner we use a compatible Bazel version
- Created Bazel branch that adds interface extensions (interfaceNames) to Bazel's Jacoco implementation
- Java 8 needs to be used for building Jacoco

### Description
This fixes the build issues reported in #1483.

### CODE REVIEW

File: scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh [hunk 1]
```diff
@@ -36,6 +36,10 @@
 
 set -e
 
+# Ensure Java 8 is used for building Jacoco (experienced issue when using e.g. Java 17).
+# You may need to change this on your system.
+export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+export PATH=$JAVA_HOME/bin:$PATH
```
Comment (tanishiking) [inline scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh:42]: I think these settings are too specific (I have different place for JAVA_HOME, and I can switch java versions using jenv in my local environment).

What do you think to add a sentence "Ensure Java 8 is used for building Jacoco" in `docs/coverage.md` and remove these `export`s?

File: scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh [hunk 2]
```diff
@@ -108,7 +114,9 @@ mvn clean install
 (
 cd $bazel_repo
 git remote update
-git checkout tags/$bazel_tag
+git checkout $bazel_branch
+
+echo "$bazel_version" > .bazelversion
```
Comment (tanishiking) [inline scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh:119]: :+1:

**Review by tanishiking — APPROVED** [review]
Hey, @gergelyfabian thank you so much for fixing both this script and the branch!
Looks like we still miss some synthetic methods generated by case class, we can build the runner with the script, and it looks better :tada:

Screenshot 2023-03-22 at 13 34 12
tested on https://github.com/tanishiking/bazel-playground/tree/main/17-scala-coverage

File: scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh (hunk 1 above)
Comment (gergelyfabian) [reply scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh:42]: Whatever solution works for you. Should I comment these lines and leave a fixme?

File: scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh (hunk 2 above)
Comment (gergelyfabian) [reply scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh:119]: This is fixing Bazel autodownloading version 6+ when building JacocoRunner, that breaks that build...

**Comment by gergelyfabian** [comment]
@liucijus I have no idea why the automatic CI build fails. Do you have any idea? I just changed a script that should not be used in the CI build...

File: scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh (hunk 1 above)
Comment (gergelyfabian) [reply scripts/build_jacocorunner/build_jacocorunner_bazel_5.0+.sh]: Updated my commit. Now, you need to manually ensure, and the script will error out if the wrong Java version is set.

**Review by liucijus — APPROVED** [review]
Thanks a lot, @gergelyfabian!

**Pull request merged**